### PR TITLE
Fix tests, update Rasterio

### LINKED ISSUE #24: Tests broken on master
Tests fail on Python 3.6 - see [this Travis run](https://travis-ci.org/mapbox/untiler/builds/229187081).

<details>
<summaryFailure</summary>
<pre>
    def test_extract_mbtiles():
        with TestTiler() as tt:
            testpath = tt.path
            testmbtiles = os.path.join(os.path.dirname(__file__), 'fixtures/testtiles.mbtiles')
            runner = CliRunner()
            result = runner.invoke(cli, [
                'streammbtiles', testmbtiles, testpath, '-z', '16', '-x', '-s',
                '{z}-{x}-{y}-mbtiles.tif', '--co', 'compress=lzw'])
            assert result.exit_code == 0
            expected_checksums = [[13858, 8288, 51489, 31223], [17927, 52775, 411, 9217]]
            for o, c in zip(result.output.rstrip().split('\n'), expected_checksums):
                with rio.open(o) as src:
                    checksums = [src.checksum(i) for i in src.indexes]
>                   assert checksums == c
E                   assert [17927, 52775, 411, 9217] == [13858, 8288, 51489, 31223]
E                     At index 0 diff: 17927 != 13858
E                     Use -v to get the full diff
</pre>
</details>

<br />

Noticed this while working on the [autodeploy branch](https://github.com/mapbox/untiler/tree/autodeploy). Initially thought this was failing due to changes I'd made, but I think this is some sort of config/dependency rot that's happening otherwise.

### PR DESCRIPTION
**Pull request by bhavika**
Changes:

- Replaces libgdal1h with libgdal1i in the Travis build
- Updates rasterio from 1.0a8 to 1.1.2 
- Instead of checking exit codes, tests now explicitly check for the type of exception raised
- Fixes #24 
- Drops Python 2.7 Travis builds

### CODE REVIEW

**Comment by bhavika** [comment]
@jacquestardie I'm most worried about whether dropping Py 2.7 builds is the right thing to do here.

File: .travis.yml [hunk 1]
```diff
@@ -21,13 +21,12 @@ env:
...
   - "3.6"
 
 addons:
   apt:
     packages:
-    - libgdal1h
+    - libgdal1i
     - gdal-bin
     - libgdal-dev
     - libatlas-dev
```
Comment (sgillies) [inline .travis.yml]: @bhavika unless untiler makes use of the gdal-bin programs, I think our CI can use a rasterio wheel with GDAL included (same for numpy and libatlas-dev, &c) and all of these apt packages can go.

File: .travis.yml [hunk 2]
```diff
@@ -21,13 +21,12 @@ env:
     - PIP_FIND_LINKS=file://$HOME/.cache/pip/wheels
 
 python:
-  - "2.7"
```
Comment (sgillies) [inline .travis.yml]: Yes, we should drop 2.7. Maybe add 3.7 instead to prepare for the future?

File: .travis.yml (hunk 1 above)
Comment (bhavika) [reply .travis.yml]: @sgillies oh, thanks for explaining that! I can remove these then, I dont think untiler uses gdal-bin.

Comment (sgillies) [reply .travis.yml]: You're welcome. This project's config could become as simple as the one in https://github.com/mapbox/rio-color/blob/master/.travis.yml.

**Review by jqtrde — APPROVED** [review]

**Comment by sgillies** [comment]
@bhavika I see the cause of the failure with 3.7: the expression patterns in `parse_template()` (see https://travis-ci.org/mapbox/untiler/jobs/653117987#L956) and probably elsewhere need to be changed to raw strings. `r"\d+"` and not "\d+".

**Comment by bhavika** [comment]
> @bhavika I see the cause of the failure with 3.7: the expression patterns in `parse_template()` (see https://travis-ci.org/mapbox/untiler/jobs/653117987#L956) and probably elsewhere need to be changed to raw strings. `r"\d+"` and not `"\d+"`. https://docs.python.org/3/howto/regex.html#the-backslash-plague

Thank you! I would have chased my tail for a while if you hadn't pointed that out. :D

File: untiler/scripts/tile_utils.py [hunk 3]
```diff
@@ -148,7 +148,7 @@ def parse_template(template):
         if len(separator) != 2 or separator[0] != separator[1]:
             raise ValueError('Too many / not matching separators!')
     
-        return valPattern.sub('\d+', template), valPattern.sub('%s', template), separator[0]
+        return valPattern.sub(r"\\d+", template), valPattern.sub("%s", template), separator[0]
```
Comment (sgillies) [inline untiler/scripts/tile_utils.py:151]: oh I get it, because we're using re to parse an re expression :joy:

**Pull request merged**